### DISCUSSION TITLE
widget_styles in modes don't work

### DISCUSSION THREAD

**Discussion by gbouthenot** [missionpinball/missionpinball#114] [Q&A]
Hello.
I am trying to define some widget_styles inside a mode, but it appears that MPF, does not see styles defined in mode.
If I put the widget_styles in the machine config, it works.

File `/modes/base/config/base.yaml`:
```#config_version=5

mode:
  start_events: ball_started
  priority: 100

widget_styles:
  ws_ball:
    x: 10
    y: 10

slides:
  sl_main_score:
    widgets:
      - type: text
        text: "menu_ball1"
        style: ws_ball

slide_player:
  mode_base_started: sl_main_score
```

This throws the error:
```
  File "/home/gilles/.pyenv/versions/3.9.7/envs/mpf-056/lib/python3.9/site-packages/mpfmc/uix/widget.py", line 386, in _apply_style
    raise ValueError("{} has an invalid style name: {}".format(
ValueError: <Text Widget text=> has an invalid style name: 'ws_ball'
```

However, if I move `widget_styles` into the file `config/config.yaml`, my `ws_ball` style works.

The documentation https://missionpinball.org/config/widget_styles/ says that `widget_styles` can be used in modes, and indeed, the declaration seems valid (mpf does not crash), but the defined styles are not found when I use them in a widget.

Versions: 
mpf                   0.56.2
mpf-mc                0.56.1

**Comment by atummons** [comment]
From looking at actual code it is only supported in config. There are some commented out tests that speak to future support for styles inside a mode.

Is this a meaningful feature to add? Or just confusion given the incorrect documentation?

**Reply by gbouthenot** [reply]
Yes, it would reduce clutter, allowing to have widget_styles defined in the mode config file, and also used in the file.
Currently, I had to copy the widget_styles that I use only in a specific mode, in the machine config.
Having widget_styles in the mode would be more elegant.

Thank you for your reply !
